### DISCUSSION TITLE
Problem connecting via Domain or Public IP

### DISCUSSION THREAD

**Discussion by durbanbr** [rustdesk/rustdesk#2220] [General]
Hi,
I purchased a Domain and put it right on my firewall, for my internal server. When I try to access it, I can't connect, it fails, but it gets "ready", that is, it has communication with the server. When I put the Internal IP, directly on the server, I can connect to the remote machine. What configuration do I need to make for it to communicate? 

I installed the server yesterday, so it's more up to date. I installed it using the github script. The server is installed on Debian 11 and the clients are Windows 11 machines

**Comment by CW3MH6** [comment]
Are you on the same network as your server, and trying to hit your server from your external IP/domain? Did you forward all relevant ports (21116-21119 udp/tcp)?

NAT reflection/loopback may not be supported/setup on your firewall/router. Please see [this link about NAT hairpinning](https://en.wikipedia.org/wiki/Network_address_translation#NAT_hairpinning)

If this is your problem and you cannot configure your firewall, the easiest fix is to add the domain to your hosts file (C:\Windows\system32\drivers\etc\hosts on Windows, /etc/hosts on Linux) and specify your domain and the internal IP of your server. So when you go to mydomain.com it will point to 192.X.X.X instead of your public IP.

**Comment by durbanbr** [comment]
Thanks CW3MH6 for the reply.

Our firewall is from Forticlient and we have already done all manual releases:

ufw allow 21115:21119/tcp
ufw allow 8000/tcp
ufw allow 21116/udp

When we make the communication, we see the request coming through port 21116 UDP, but I don't see the return. The configuration part in the firewall is all correct. Is there something I need to release in addition to these ports or is some configuration wrong?

**Comment by pedjas** [comment]
I met some issues with server behind NAT that may be similar to yours. Check my solution https://github.com/rustdesk/rustdesk/discussions/2152
